flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -81,25 +134,51 @@ public void configure(Configuration parameters) {
 	 * @throws IOException
 	 */
 	@Override
-	public void open(InputSplit ignored) throws IOException {
+	public void open(InputSplit inputSplit) throws IOException {
+		hasNext = true;
```

Comment:
Call `resultSet.next()` at the end of `open()` and only set `hasNext = true` if the split actually contains a record.